src/server/external-services/pathway-commons.js

```diff
@@ -17,74 +17,87 @@ const fetchOptions = {
   }
 };
 
-const _sanitize = (s) => {
+//Pathway Commons HTTP GET request; options.cmd = 'pc2/get', 'pc2/search', 'pc2/traverse', 'pc2/graph', etc.
+let query = opts => {
+  let queryOpts = _.assign( { user: 'app-ui', cmd: 'pc2/get' }, opts);
+  let { cmd } = queryOpts;
+  let url = config.PC_URL + cmd + '?' + qs.stringify( queryOpts );
+
+  return fetch(url, fetchOptions)
+    .then(res => ( cmd === 'pc2/get' || cmd === 'pc2/graph' ? res.text() : res.json() ) )
+    .catch((e) => {
+      logger.error('query ' + queryOpts + ' failed - ' + e);
+      return null;
+    });
+};
+
+let sanitize = s => {
   // Escape (with '\'), to treat them literally, symbols, such as '*', ':', or space,
   // which otherwise play special roles in a Lucene query string.
   return s.replace(/([!*+\-&|()[\]{}^~?:/\\"\s])/g, '\\$1');
 };
 
-const _processPhrase = (phrase) => {
-  return validatorGconvert(phrase.split(' '),{}).then(result => {
+
+// recognize biological entities from an input string
+let extractEntityIds = inputString => {
+  let tokens = inputString.split(' ');
+
+  return validatorGconvert( tokens ).then( result => {
     let { unrecognized, alias  } = result;
-    const entities = _.keys( alias ).map( initialAlias =>'xrefid:' + _sanitize( initialAlias.toUpperCase()) );
-    const otherIds = unrecognized.map(id=>{
-      id=id.toUpperCase();
-      const recognized = /^SMP\d{5}$/.test(id) // check for a smpdb or chebi id
-        ||/^CHEBI:\d+$/.test(id) && (id.length <= ("CHEBI:".length + 6));
-      const sanitized = _sanitize(id);
+    let entities = _.keys( alias ).map( initialAlias => 'xrefid:' + sanitize( initialAlias.toUpperCase() ) );
+
+    let otherIds = unrecognized.map( id => {
+      id = id.toUpperCase();
+      let isChebiId = /^CHEBI:\d+$/.test( id );
+      let isSmpdbId = /^SMP\d{5}$/.test( id );
+      let recognized = isSmpdbId || isChebiId && ( id.length <= ("CHEBI:".length + 6) );
+      let sanitized = sanitize(id);
+
       return recognized ? ( 'xrefid:' + sanitized ) : ( 'name:' + '*' + sanitized + '*' );
     });
+
     return entities.concat(otherIds);
+  })
+  .catch( e => {
+    logger.error('unable to get response from gconvert with the following inputstring: ' + inputString);
+    logger.error(e);
+    return [];
```

Comment:
Not sure if it is a good idea to return stuff from a catch.

Reply:
If validator fails, I was thinking to return, for each item in `tokens`:
 `'name:' + '*' + sanitized + '*'`

Reply:
ok

Reply:
It may be better to recover from the error closer to the UI, either in the UI directly (ideal) or in a higher level service that goes through all the pipeline steps.

Reply:
I think this is part of the search service pipeline.